File: prototype/rails/src/app/delivery/flash_magic_link_delivery.rb
class FlashMagicLinkDelivery
  def initialize(flash)
    @flash = flash
  end

  # The debug-alert partial in both layouts renders whatever lands under this
  # key, which is how the prototype hands over a link with no mailbox.
  def deliver(email:, url:)
    @flash[:debug_magic_link] = url
  end
end

File: prototype/node/src/app/delivery/flash-magic-link-delivery.ts
import type { Flash } from '../plugins/flash.ts'
import type { MagicLinkDelivery, MagicLinkMessage } from './magic-link-delivery.ts'

/**
 * Hands the link to the debug alert every layout renders, which is how the
 * prototype delivers a sign-in link with no mailbox anywhere.
 */
export const flashMagicLinkDelivery: MagicLinkDelivery = {
  deliver(message: MagicLinkMessage): Flash {
    return { debugMagicLink: message.url }
  },
}
